params.go

```diff
@@ -72,6 +73,14 @@ type Params struct {
 	RollingUpdate          RollingUpdateParams       `json:"rollingupdate,omitempty" yaml:"rollingupdate,omitempty"`
 }
 
+type Toleration struct {
```

Comment:
Although strongly typing the tolerations is useful for validating them it does make it harder to only output the final yaml with just the properties the user want to use. A toleration doesn't always need `key` and `value` for example.

An alternative would be what we do for `initContainers`, where 

- the param is of type `[]*map[string]interface{}` - which allows any yaml to be used in the manifest - see https://github.com/estafette/estafette-extension-gke/blob/eae31c3a482dbb9fc6494bd870b4413cc4c1e07c/params.go#L68.
- and the template renders it to yaml again, see https://github.com/estafette/estafette-extension-gke/blob/main/templates/deployment.yaml#L70-L73

We can probably do this for more parts where we have exactly the same properties as in the final yaml manifest and don't check anything before testing it against the kubernetes server.